ziti/cmd/ziti/cmd/config_templates/router.yml

```diff
@@ -0,0 +1,90 @@
+{{/*
+ Config Format Version
+
+ Whenever a breaking change is made to the semantics of this configuration file, the configuration version
+ expected by the router will be incremented. When the controller tries to load a configuration file with
+ an incompatible version, it will abort with a message prompting the operator to seek out the breaking changes
+ documentation.*/}}
+v: 3
+
+identity:
+  cert:                 "{{ .ZitiPKI }}/certs/{{ .Router.Name }}-client.cert"
+  server_cert:          "{{ .ZitiPKI }}/certs/{{ .Router.Name }}-server.cert"
+  key:                  "{{ .ZitiPKI }}/keys/{{ .Router.Name }}-server.key"
+  ca:                   "{{ .ZitiPKI }}/certs/{{ .Router.Name }}-cas.cert"
+
+ctrl:
+  endpoint:             tls:{{ .Controller.Hostname }}:{{ .Controller.FabCtrlPort }}
+
+link:
+{{ if .Router.IsPrivate }}#{{ end }}  listeners:
+{{ if .Router.IsPrivate }}#{{ end }}    - binding:          transport
+{{ if .Router.IsPrivate }}#{{ end }}      bind:             tls:0.0.0.0:{{ .Router.Listener.BindPort }}
+{{ if .Router.IsPrivate }}#{{ end }}      advertise:        tls:{{ .Router.Edge.Hostname }}:{{ .Router.Listener.BindPort }}
+{{ if .Router.IsPrivate }}#{{ end }}      options:
+{{ if .Router.IsPrivate }}#{{ end }}        outQueueSize:   {{ .Router.Listener.OutQueueSize }}
+  dialers:
+    - binding: transport
+
+listeners:
+{{ if or .Router.IsFabric .Router.IsPrivate }}#{{- end }}  - binding: tunnel
+{{ if or .Router.IsFabric .Router.IsPrivate }}#{{- end }}    options:
+{{ if or .Router.IsFabric .Router.IsPrivate }}#{{- end }}      mode: host #tproxy|tun|host
+{{ if .Router.IsPrivate }}#{{ end }}  - binding: {{ if or .Router.IsFabric .Router.IsPrivate }}transport{{ else }}edge{{ end }}
+{{ if .Router.IsPrivate }}#{{ end }}    address: {{ if .Router.IsWss }}ws{{ else }}tls{{end}}:0.0.0.0:{{ if .Router.IsWss }}3023{{ else }}{{ .Router.Edge.Port }}{{ end }}
+{{ if .Router.IsPrivate }}#{{ end }}    options:
+{{ if .Router.IsPrivate }}#{{ end }}      advertise: {{ .Router.Edge.Hostname }}:{{ if .Router.IsWss }}3023{{ else }}{{ .Router.Edge.Port }}{{ end }}
+{{ if .Router.IsPrivate }}#{{ end }}      connectTimeoutMs: {{ .Router.Listener.ConnectTimeoutMs }}
+{{ if .Router.IsPrivate }}#{{ end }}      getSessionTimeout: {{ .Router.Listener.GetSessionTimeoutS }}s
+
+
+{{- if or .Router.IsPrivate .Router.IsFabric }}
+#edge:
+csr:
+  country: US
+  province: NC
+  locality: Charlotte
+  organization: NetFoundry
+  organizationalUnit: Ziti
+  sans:
+    dns:
+      - {{ .Router.Edge.Hostname }}
```

Comment:
one is probably fine but if there were a way to take in an array that'd be better. something to think about later on maybe. i'd rename this to "ExternalAddress" maybe.